### PR TITLE
Log EOFs found on Kubernetes watcher at debug level

### PR DESCRIPTION
**Pull request by jsoriano**
EOFs received when watching Kubernetes events can be recovered by
reconnecting, and they don't need to be harmful. So log them at the
debug level.
If after the EOF there are still problems, errors will still be logged at
the error level when reconnecting.

### CODE REVIEW

File: libbeat/common/kubernetes/watcher.go [hunk 1]
```diff
@@ -255,12 +255,17 @@ func (w *watcher) watch() {
...
 			eventType, err := watcher.Next(r)
 			if err != nil {
-				logp.Err("kubernetes: Watching API error %v", err)
 				watcher.Close()
-				if !(err == io.EOF || err == io.ErrUnexpectedEOF) {
+				switch err {
+				case io.EOF:
+					logp.Debug("kubernetes", "EOF while watching API")
+				case io.ErrUnexpectedEOF:
+					logp.Warn("kubernetes: Unexpected EOF while watching API")
```
Comment (ruflin) [inline libbeat/common/kubernetes/watcher.go]: Historically we tried not to use `Warn` but Error or Info to make it clear if action is needed or not. I'm tempted to think this should be logged on `Info` level as no action is needed normally?

Comment (jsoriano) [reply libbeat/common/kubernetes/watcher.go]: Yeah, changed to `Info` :+1:

**Comment by ruflin** [comment]
jenkins, test this

**Comment by ruflin** [comment]
I have seen the elasticsearch tests failing now on 2-3 PR's. I wonder if there was a change in ES that is causing it. Will investigate if it happens again.

**Comment by jsoriano** [comment]
@ruflin changelog entry added

**Pull request merged**